SETUP_KSIM/KSIM_INPUT_PARAMETERS.txt
object_name = Standard_Star_GD71                               
object_file = GD71_spec.txt       
object_x = Wavelength / $nm$
object_y = Flux / $ergcm^{-2}s^{-1}\AA^{-1}$ 
binstep = 0.006266605912991086                 
mirr_diam = 800        
seeing = 0.8         
h = 6.626e-27 
c = 3e10 
exposure_t = 30                    
tele_file = gemini_mirrors_Ag_coating_data_nm                
lambda_low_val = 350          
lambda_high_val = 1900          
n_pixels = 3800                  
alpha_val = 25.0             
phi_val = 25.0     
opt_refl_deg = 6.03                                                                   
opt_grooves = 173.8                                                                           
norders = 35 
folder_dir = STANDARD_STAR  
fudicial_energy_res = 40.    
fudicial_wavelength = 374.                                        
raw_sky_file = skytable2.fits       
slit_width = 0.6                      
pixel_fov = 0.3           
off_centre = 0.0                    
airmass = 1.5                      
ir_grooves = 110.0                               
ir_alpha_val = 30.0             
ir_phi_val = 30.0   
ir_refl_deg = 7.6          
dead_pixel_perc = 0.0         
R_E_spread = False                         
IR_arm = False  
cutoff = 1900 
TLUSTY = False     
object_file_1 = T498g775_2_34kpc_78_steps 
object_file_2 = T100g825_2_34kpc_78_steps 
row = 20 
redshift = 0.0 
redshift_orig = 0.200344 
mag_reduce = 1.0        
sky = True   
generate_sky_seeing_eff = False      
sky_seeing_eff_file_save_or_load = SKY_SEEING_21072022_300_1900nm_800cmtele      
generate_model_seeing_eff = False                                  
model_seeing_eff_file_save_or_load = GD71_800cmtele    
generate_additional_plots = False                      
generate_standard_star_factors = False                       
stand_star_run_filename_details = _1arm_30Re_5000R_50cmtele                
supported_file_extraction = False                
fwhm_fitter = False           
fwhm_fitter_central_wavelength = 656
fwhm_fitter_central_wavelength_2 = 729  
double_fitter = False   
continuum_removal_use_polynomial = True    
reset_R_E_spread_array = False                 
reset_dead_pixel_array = False           
reg_grid_factor = 5          

SETUP_KSIM/KSIM_INPUT_PARAMETERS_l.txt
object_name = Standard_Star_GD71                               
object_file = GD71_spec.txt       
object_x = Wavelength / $nm$
object_y = Flux / $ergcm^{-2}s^{-1}\AA^{-1}$ 
binstep = 0.006266605912991086                 
mirr_diam = 800        
seeing = 0.8         
h = 6.626e-27 
c = 3e10 
exposure_t = 30                    
tele_file = gemini_mirrors_Ag_coating_data_nm                
lambda_low_val = 350          
lambda_high_val = 1900          
n_pixels = 4000                  
alpha_val = 25.0             
phi_val = 25.0     
opt_refl_deg = 10.75                                                                   
opt_grooves = 278.0                                                                           
norders = 35 
folder_dir = STANDARD_STAR  
fudicial_energy_res = 25.    
fudicial_wavelength = 380.                                        
raw_sky_file = skytable2.fits       
slit_width = 0.6                      
pixel_fov = 0.3           
off_centre = 0.0                    
airmass = 1.5                      
ir_grooves = 110.0                               
ir_alpha_val = 30.0             
ir_phi_val = 30.0   
ir_refl_deg = 7.6          
dead_pixel_perc = 0.0         
R_E_spread = False                         
IR_arm = False  
cutoff = 1900 
TLUSTY = False     
object_file_1 = T498g775_2_34kpc_78_steps 
object_file_2 = T100g825_2_34kpc_78_steps 
row = 20 
redshift = 0.0 
redshift_orig = 0.200344 
mag_reduce = 1.0        
sky = True   
generate_sky_seeing_eff = False      
sky_seeing_eff_file_save_or_load = SKY_SEEING_21072022_300_1900nm_800cmtele      
generate_model_seeing_eff = False                                  
model_seeing_eff_file_save_or_load = GD71_800cmtele    
generate_additional_plots = False                      
generate_standard_star_factors = False                       
stand_star_run_filename_details = _1arm_30Re_5000R_50cmtele                
supported_file_extraction = False                
fwhm_fitter = False           
fwhm_fitter_central_wavelength = 656
fwhm_fitter_central_wavelength_2 = 729  
double_fitter = False   
continuum_removal_use_polynomial = True    
reset_R_E_spread_array = False                 
reset_dead_pixel_array = False           
reg_grid_factor = 5          

parameters.py
# -*- coding: utf-8 -*-
"""
Created on Tue Nov 19 15:01:39 2019

@author: BVH
"""

import numpy as np
import os

'''
This file stores all the parameters for the simulation.
For all file names don't forget to include the path if the file is not in 
the same directory as the SIM scripts.
'''

def str_2_bool(val):
    if val == 'True':
        return True
    elif val == 'False':
        return False
    else:
        raise ValueError("Accepted inputs are True or False")
        

parameter_file = 'KSIM_INPUT_PARAMETERS.txt'

file_param = open('SETUP_KSIM/KSIM_INPUT_PARAMETERS.txt','r')

params = file_param.readlines()

file_param.close()

folder = params[19].split(' ')[2]

if os.path.isdir(folder) == False:
    os.mkdir(folder)

object_name = params[0].split(' ')[2]
object_file = params[1].split(' ')[2]
object_x = params[2].split(' ')[2:5]#what will be the xlabel of the spectrum graph
object_y = params[3].split(' ')[2:5] #what will be the ylabel of the spectrum graph

object_x = object_x[0] + ' ' + object_x[1] + ' ' + object_x[2]
object_y = object_y[0] + ' ' + object_y[1] + ' ' + object_y[2]


binstep = float(params[4].split(' ')[2]) * 1e-7
mirr_diam = float(params[5].split(' ')[2]) #diameter of the VLT primary mirror(in cm)
seeing = float(params[6].split(' ')[2]) #again working number, will depend on the telescope being used
h = float(params[7].split(' ')[2]) # in ergs/s
c = float(params[8].split(' ')[2]) #in cm/s
exposure_t = float(params[9].split(' ')[2])

fov = 30.0*60.0
area = (0.5*mirr_diam)**2.

tele_file = params[10].split(' ')[2]

lambda_low_val = float(params[11].split(' ')[2]) #in nm
lambda_high_val = float(params[12].split(' ')[2]) #in nm
n_pixels = int(params[13].split(' ')[2])
alpha_val = float(params[14].split(' ')[2]) # Angle of incidence in degrees
phi_val = float(params[15].split(' ')[2])  # Grating blaze angle in degrees
refl_deg_opt = float(params[16].split(' ')[2]) 
refl_start_val = alpha_val - refl_deg_opt # Diffraction angles in degrees
refl_end_val = alpha_val + refl_deg_opt    # Beta start and stop NEED NOT be symmetric around alpha in degrees
OPT_grooves = float(params[17].split(' ')[2])
norders = float(params[18].split(' ')[2])
OPT_a = 1000. / OPT_grooves #distance between grooves in micrometers
ER_band_low = float(params[20].split(' ')[2]) #energy resolution of MKID pixels for low wavelength
ER_band_wl = float(params[21].split(' ')[2]) 

raw_sky_file = params[22].split(' ')[2] #sky photon counts

slit_width = float(params[23].split(' ')[2])
pix_fov = float(params[24].split(' ')[2]) #pixel fov

eff_models = ['Gaussian','Sinc','Schroeder','Casini'] #grating efficiency model names

off_centre = float(params[25].split(' ')[2]) #how much the object is not in the centre, between 0 and 1
airmass = float(params[26].split(' ')[2])

IR_grooves = float(params[27].split(' ')[2])
IR_a = 1000. / IR_grooves #distance between grooves in micrometers
IR_alpha = float(params[28].split(' ')[2])
IR_phi = float(params[29].split(' ')[2])
refl_deg_ir = float(params[30].split(' ')[2]) 
refl_start_val_ir = IR_alpha - refl_deg_ir #6.0 # Diffraction angles in degrees
refl_end_val_ir = IR_alpha + refl_deg_ir #6.0  # Beta start and stop NEED NOT be symmetric around alpha in degrees

dead_pixel_perc = float(params[31].split(' ')[2])
r_e_spread = str_2_bool(params[32].split(' ')[2])

IR_arm = str_2_bool(params[33].split(' ')[2])
cutoff = float(params[34].split(' ')[2])
TLUSTY = str_2_bool(params[35].split(' ')[2])
object_file_1 = params[36].split(' ')[2]
object_file_2 = params[37].split(' ')[2] 
row = int(params[38].split(' ')[2])
redshift = float(params[39].split(' ')[2])
redshift_orig = float(params[40].split(' ')[2])
mag_reduce = float(params[41].split(' ')[2])
sky = str_2_bool(params[42].split(' ')[2])
gen_sky_seeing_eff = str_2_bool(params[43].split(' ')[2])
sky_seeing_eff_file_save_or_load = params[44].split(' ')[2]
gen_model_seeing_eff = str_2_bool(params[45].split(' ')[2])
model_seeing_eff_file_save_or_load = params[46].split(' ')[2]
extra_plots = str_2_bool(params[47].split(' ')[2])
stand_star_factors_run = str_2_bool(params[48].split(' ')[2])
stand_star_filename_detail = params[49].split(' ')[2]
supported_file_extraction = str_2_bool(params[50].split(' ')[2])
fwhm_fitter = str_2_bool(params[51].split(' ')[2])
cen_wl = float(params[52].split(' ')[2])
cen_wl_2 = float(params[53].split(' ')[2])
double_fit = str_2_bool(params[54].split(' ')[2])
cont_rem_poly = str_2_bool(params[55].split(' ')[2])
reset_R_Es = str_2_bool(params[56].split(' ')[2])
reset_dead_pixels = str_2_bool(params[57].split(' ')[2])
reg_grid_factor = float(params[58].split(' ')[2])

pix_mult = np.ndarray.astype((np.linspace(1,n_pixels,10)),dtype='int') #for progress updates during sections of the code involving looping through pixels
 
seeing_str = str(seeing)[0] + '_' + str(seeing)[2]
#refl_start_val = alpha_val - ((n_pixels*pix_fov)/3600) # Diffraction angles in degrees
#refl_end_val = alpha_val + ((n_pixels*pix_fov)/3600) 

slit_R_factor = slit_width/pix_fov


